runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldSwitcherPanel.java

```diff
@@ -159,24 +159,24 @@ void updateList()
 			switch (orderIndex)
 			{
 				case PING:
-					return Integer.compare(r1.getPing(), r2.getPing()) * (ascendingOrder ? 1 : -1);
+					// Leave worlds with unknown ping at the bottom
+					return r1.getPing() > 0 == r2.getPing() > 0
+						? Integer.compare(r1.getPing(), r2.getPing()) * (ascendingOrder ? 1 : -1)
+						: r1.getPing() <= 0 ? 1 : -1;
 				case WORLD:
 					return Integer.compare(r1.getWorld().getId(), r2.getWorld().getId()) * (ascendingOrder ? 1 : -1);
 				case PLAYERS:
 					return Integer.compare(r1.getUpdatedPlayerCount(), r2.getUpdatedPlayerCount()) * (ascendingOrder ? 1 : -1);
 				case ACTIVITY:
-					return r1.getWorld().getActivity().compareTo(r2.getWorld().getActivity()) * -1 * (ascendingOrder ? 1 : -1);
+					// Leave empty activity worlds on the bottom of the list
+					return r1.worldHasActivity() == r2.worldHasActivity()

```

Comment:
This should be abstracted out into logical if statements IMO instead of nested ternaries.